### PR TITLE
Rob10470/Update runtimecore to 68

### PR DESCRIPTION
**Pull request by robert-craig-houston**
Updates the runtimecore branch from ICU63 to ICU68 (Taken from the ICU maint68 branch) while including only the previously required files and data. Data has been recompiled as expected using the new filter based system, with a similar sized result as the manual deletion method. Have already tested on a seperate branch (But will rerun all tests of course!). The upgrade overall slightly reduces the size of runtime, somewhat significantly in the case of android!

Upgrading ICU is desireable for many reasons (such as improved emoji support + security concerns) but the primary driver are requests for date and number internationlization options for converting numbers and dates to text in Arcade, and ICU is ideal to support this as it should enable parity between the C++ and JS behaviours. Access to date functionality that has historically been a source of many bugs inside the expression library and has been hit elsewhere is an added bonus. Keeping ICU up to date will enable this.

After this initial upgrade we can examine the data that would be needed to resolve the internationlization functions and other longstanding bugs, and work out a suitable solution. 

Further details can be found in the issues below below: 
Expression issue to upgrade ICU: https://devtopia.esri.com/runtime/arcade/issues/719
Expression issue for arcade internationalization functions: https://devtopia.esri.com/runtime/arcade/issues/663
Release plan issue regarding Arcade Internationalization functions: https://devtopia.esri.com/runtime/runtime-sdk-release-plan/projects/10#card-89598
Release plan issue regarding upgrading ICU: https://devtopia.esri.com/runtime/runtime-sdk-release-plan/projects/10#card-89615

All tests were performed on the same branch of runtimecore master, with all factors other than ICU the same. The change in binary size for the compiled ICU Library and runtime are as follows:

| size in kb | ICU63 | ICU68 | Size in kb delta | Percentage of new size relative to old. |
|------------|-------|-------|------------------|-----------------------------------------|
| Android    | 5659  | 3597  | -2062            | 64%                                     |
| Cocoa      | 5609  | 3524  | -2085            | 63%                                     |
| IOS        | 5480  | 3440  | -2040            | 63%                                     |
| Linux      | 5688  | 3614  | -2074            | 64%                                     |
| Macos      | 5530  | 3570  | -1960            | 65%                                     |
| Windows    | 37016 | 43275 | 6259             | 117%                                    |
| Winuwp     | 46000 | 43647 | -2353            | 95%                                     |

size in kb | Runtime63 | runtime68 | Size in kb delta | Percentage of new size   relative to old.
-- | -- | -- | -- | --
Android | 43670 | 40480 | -3190 | 93%
Cocoa | ??? | 41455 | NA | NA
IOS | 400070 | 395900 | -4170 | 99%
Linux | 103641 | 101449 | -2192 | 98%
Macos | 120140 | 118010 | -2130 | 98%
Windows | 85710 | 83474 | -2236 | 97%
Winuwp | 76791 | 74561 | -2230 | 97%

Was unable to find the Cocoa build size for the runtimecore using 63, but given there were no errors and similar projects have shown minor size improvements any issues are unlikely. These changes in size are likely partially due to the changed data build method ommitting more of the uneeded files, as well as improvements to ICU's build scripts.

### CODE REVIEW

**Comment by robert-craig-houston** [comment]
@chri7325 @cwarner-8702 
Hi Christian and Chris, Sorry to bother you both! The two of you have had numerous commits into the runtimecore branch and thought you'd be the best placed to review this update to 68. If you can think of anyone else that should be pinged, or better practice to avoid pinging you both do let me know! 

Thanks for your help, apologies for any hassle!

File: icu.lua [hunk 1]
```diff
@@ -5,19 +5,22 @@ dofile(_BUILD_DIR .. "/static_library.lua")
...
 defines {
-  -- Limit the number of conversions supported to only those necessary: ASCII, Latin-1, and
-  -- UTF-8/UTF-16.  See ucnv_bld.cpp for a list of encodings available with each #define
-  "UCONFIG_NO_LEGACY_CONVERSION",
-  "UCONFIG_ONLY_HTML_CONVERSION",
+    -- Limit the number of conversions supported to only those necessary: ASCII, Latin-1, and
+    -- UTF-8/UTF-16.  See ucnv_bld.cpp for a list of encodings available with each #define
+    "UCONFIG_NO_LEGACY_CONVERSION",
+    "UCONFIG_ONLY_HTML_CONVERSION",
+	"UCONFIG_NO_SERVICE",
```
Comment (cwarner-8702) [inline icu.lua:13]: Inconsistent indentation in this file

File: icu.lua [hunk 2]
```diff
@@ -120,11 +124,17 @@ if (_PLATFORM_MACOS) then
 end
 
 if (_PLATFORM_WINDOWS) then
+  buildoptions {
+  "/utf-8",
```
Comment (cwarner-8702) [inline icu.lua:128]: This should be indented more than the `buildoptions` tag.

**Review by cwarner-8702 — CHANGES_REQUESTED** [review]

**Review by cwarner-8702 — APPROVED** [review]
LGTM.  Make sure to wait for @chri7325 to weigh in

**Comment by robert-craig-houston** [comment]
Will do, Thanks for the guidance!

**Review by chri7325 — CHANGES_REQUESTED** [review]
I think this branch may need to be recreated actually. The PR can remain but looking at how the commits were brought in, it doesn't look like a merge with the ICU release branch but rather, a squashed merge. We try to preserve the history of branches so it is easy to do future upgrades and diff with the upstream source. Otherwise, things look good.

**Comment by chri7325** [comment]
> Android | 43670 | 40480 | -3190 | 93%

The android change may actually be a bit deceiving here. The android build had to remove unity builds, due to a KML bug so the loss in size is probably more due to the daily build being a regular and the vTest run being unity. Still great to see a drop in size though!

**Pull request closed without merging**